warp10/src/main/java/io/warp10/json/JsonUtils.java

```diff
@@ -57,33 +52,35 @@ public void serialize(Object value, JsonGenerator gen, SerializerProvider serial
   }
 
   /**
-   * Used to swap UnknownSerializer and BeanSerializer for CustomEncodersSerializer.
+   * Used to swap UnknownSerializer and BeanSerializer for CustomSerializer.
    */
   public static class NotSerializedToCustomSerializedModifier extends BeanSerializerModifier {
     @Override
     public JsonSerializer<?> modifySerializer(SerializationConfig config, BeanDescription beanDesc, JsonSerializer<?> serializer) {
       if (serializer instanceof UnknownSerializer || serializer instanceof BeanSerializer) {
-        return customEncodersSerializer;
+        return CUSTOM_SERIALIZER;
       } else {
         return serializer;
       }
     }
   }
 
-  public static class CustomEncodersSerializer extends JsonSerializer<Object> {
+  /**
+   * Handles custom serialization based on transformers.
+   */
+  public static class CustomSerializer extends JsonSerializer<Object> {
     @Override
     public void serialize(Object value, JsonGenerator gen, SerializerProvider serializers) throws IOException {
-      if (null != encoders && !encoders.isEmpty()) {
-        StringBuilder sb = new StringBuilder();
-        boolean encoded = false;
-        for (JsonEncoder encoder: encoders) {
-          encoded = encoder.addElement(sb, value);
-          if (encoded) {
+      if (null != transformers && !transformers.isEmpty()) {
+        JsonTransformer.TransformationResult transfRes = null;
+        for (JsonTransformer transformer: transformers) {
+          transfRes = transformer.transform(value);
+          if (null != transfRes && transfRes.transformed) {
             break;
           }
         }
-        if (encoded) {
-          gen.writeRaw(sb.toString());
+        if (null != transfRes && transfRes.transformed) {
```

Comment:
Shouldn't we leave the possibility to return JSON content too? Because the transformer may use an existing implementation of a serializer which will produce JSON. The validity of the returned JSON would be the responsability of the called entity as it was in the previous implementation.